src/main/java/org/bahmni/mart/form2/service/FormService.java

```diff
@@ -53,4 +58,31 @@
     private List<Map<String, Object>> getLatestFormNamesWithVersion() {
         return openmrsDbTemplate.queryForList("SELECT name , MAX(version) as version FROM form GROUP BY name");
     }
+
+    public Map<String, String> getFormNameTranslations(String locale) {
+        LinkedHashMap<String, String> formNameAndTranslationMap = new LinkedHashMap<>();
+        List<Map<String, Object>> forms = executeFormListQuery(form2FormNameTranslationsListResource);
+        for (Map<String, Object> form : forms) {
+            String name = (String) form.get(FORM_NAME);
+            String valueReference = (String) form.get("value_reference");
+            Gson gson = new Gson();
+            List<FormNameJsonMetadata> formNameTranslations =
+                    Arrays.asList(gson.fromJson(valueReference, FormNameJsonMetadata[].class));
+
+            String translation = locale != null ? getTranslatedFormName(locale, formNameTranslations)
+                    : getTranslatedFormName("en", formNameTranslations);
+            formNameAndTranslationMap.put(name, translation);
+        }
+        return formNameAndTranslationMap;
+
+    }
+
+    private String getTranslatedFormName(String locale, List<FormNameJsonMetadata> formNameTranslations) {
+        return formNameTranslations.stream().filter(nam -> (nam.getLocale().equals(locale)))
+                .map(FormNameJsonMetadata::getDisplay).findFirst().isPresent() ?
+                formNameTranslations.stream().filter(nam -> (nam.getLocale().equals(locale)))
+                        .map(FormNameJsonMetadata::getDisplay).findFirst().get() :
+                formNameTranslations.stream().filter(nam -> (nam.getLocale().equals("en")))
```

Comment:
If possible try to extract variables, finding it hard to read

Reply:
Sure